lib/common/errors.ts

```diff
@@ -123,14 +124,33 @@ export class Errors implements IErrors {
 
 	public printCallStack: boolean = false;
 
-	public fail(optsOrFormatStr: any, ...args: any[]): never {
-		const argsArray = args || [];
+	public fail(optsOrFormatStr: string | IFailOptions, ...args: any[]): never {
+		const opts = this.getFailOptions(optsOrFormatStr);
+		return this.failWithOptions(opts, false, args);
+	}
+
+	@deprecated("Use `fail` instead.")
+	public failWithoutHelp(optsOrFormatStr: string | IFailOptions, ...args: any[]): never {
+		return this.fail(optsOrFormatStr, args);
```

Comment:
Same as above for passing `...args` instead of `args`. It can be easily reproduced with `tns help invalidcommand` - you'll see how the error message is incorrect due to passing the full array here.